Sources/swift-format/Run.swift

```diff
@@ -28,6 +28,12 @@ func lintMain(configuration: Configuration, path: String) -> Int {
   let diagnosticEngine = makeDiagnosticEngine()
   let linter = SwiftLinter(configuration: configuration, diagnosticEngine: diagnosticEngine)
 
+  guard FileManager.default.isReadableFile(atPath: url.path) else {
```

Comment:
We should also protect ourselves at the API layer (the `SwiftLinter` and `SwiftFormatter` classes) if someone links in the formatter to another tool and passes an invalid file in the `URL` argument there.

So it might be better to put this check in there, but instead of calling `exit`, have it throw an error indicating that the file wasn't found/wasn't readable. Then, in the `do/catch` below, you can handle that error and exit gracefully.

Reply:
Done.